### ISSUE TITLE
ResponseHeaderTimeoutStage: IllegalArgumentException: toMillis not allowed on infinite Durations

### ISSUE THREAD

**Issue by fthomas** [http4s/http4s#2760]
With 0.20.8 I'm observing this exception:
```
java.lang.IllegalArgumentException: toMillis not allowed on infinite Durations
        at scala.concurrent.duration.Duration$Infinite.fail(Duration.scala:217)
        at scala.concurrent.duration.Duration$Infinite.toMillis(Duration.scala:222)
        at org.http4s.blazecore.ResponseHeaderTimeoutStage.stageStartup(ResponseHeaderTimeoutStage.scala:55)
        at org.http4s.blazecore.ResponseHeaderTimeoutStage.init(ResponseHeaderTimeoutStage.scala:60)
        at org.http4s.client.blaze.BlazeClient$.$anonfun$makeClient$21(BlazeClient.scala:128)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
        at cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
        at cats.effect.IO$.$anonfun$asyncF$1(IO.scala:1101)
        at cats.effect.IO$.$anonfun$asyncF$1$adapted(IO.scala:1093)
        at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:341)
        at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
        at cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
        at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:86)
        at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
        at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
        at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:93)
        at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:93)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
        at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:93)
        at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
        at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:44)
        at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:72)
        at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:52)
        at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
        at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:34)
        at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:44)
        at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
        at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:341)
        at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
        at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:355)
        at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:376)
        at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:316)
        at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
        at cats.effect.internals.PoolUtils$$anon$2$$anon$3.run(PoolUtils.scala:51)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```
I'm using `BlazeClientBuilder.apply` without setting any timeouts explicitly.

**Comment by fthomas** [comment]
I suspect that this exception surfaced now because `BlazeClientBuilder.responseHeaderTimeout` was set to `Duration.Inf` in https://github.com/http4s/http4s/pull/2746.

**Comment by rossabaker** [comment]
Ugh. I'd say a workaround would be to set it to an absurdly high timeout, and we'll get a patch out ASAP.

**Comment by rossabaker** [comment]
Fixed in 0.20.9.

**Comment by fthomas** [comment]
Thanks for the quick fix!
